SharedCryptoPkg/Library/CryptLibSharedDriver/PeiCryptLibSharedDriver.c

```diff
@@ -0,0 +1,78 @@
+/** @file
+SharedCryptoLibPei.c
+
+Copyright (c) 2019, Microsoft Corporation
+
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice,
+   this list of conditions and the following disclaimer.
+2. Redistributions in binary form must reproduce the above copyright notice,
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+**/
+#include <Uefi.h>
+#include <Library/BaseLib.h>
+#include <Library/DebugLib.h>
+#include <Library/PeiServicesLib.h>
+#include <Library/BaseCryptLib.h>
+#include <Ppi/SharedCryptoPpi.h>
+#include <SharedCryptoHelpers.h>
+
+SHARED_CRYPTO_FUNCTIONS *GetProtocol()
+{
+  EFI_STATUS Status;
+  UINTN Version;
+  SHARED_CRYPTO_PPI *pCryptoProt = NULL;
+  if (pCryptoProt == NULL)
+  {
+    Status = PeiServicesLocatePpi(
+        &gSharedCryptoPpiGuid,
+        0,
+        NULL,
+        (VOID **)&pCryptoProt);
+    if (EFI_ERROR(Status))
+    {
+      ProtocolNotFound(Status);
+    }
+    else {
+      Version = pCryptoProt->SharedCrypto_GetLowestSupportedVersion();
+      if (Version != SHARED_CRYPTO_VERSION)
+      {
+        ProtocolNotFound(EFI_PROTOCOL_ERROR);
+      }
+
+    }
+  }
+
+  return pCryptoProt;
+}
+
+VOID ProtocolNotFound (
```

Comment:
same bug here as dxe version if the version doesn't match you still return a pointer rathern than NULL

Reply:
Fixed

Reply:
Fixed